Reimplement ClassicQuantizeFacing using a look-up-table.

pchote opened the pull request:
This PR replaces my now-incomprehensible best-guess formula with a lookup table that matches the original game ([reference](https://github.com/TheAssemblyArmada/Chronoshift/blob/c91eb086587b895bff7fb4e53a1d31992b7bad18/src/game/engine/facing.cpp#L43-L79)). This fixes the main blocker to the Facing&rarr;WAngle upgrade, ~~and allows us to fix the facings on RA's 16-frame units.~~

penev92 commented:
Just some thoughts from a quick glance:
 - `Util` is way too generic of a name for my taste here. Something like `(Classic)FacingUtil` might be more accurate.
 - I am not convinced either the class or any of its methods need to be `public`, since they're not _really_ supposed to be used outside of that project (but there might be an edge case where external mod code could want to use them?).
 - Providing a link to the code you reference in the PR as a comment in the code might not be a bad idea. It is usually helpful to have a link to source materials inside the code (especially for such obscure stuff).

pchote commented:
For the first point: `OpenRA.Game`, `OpenRA.Mods.Common` and `OpenRA.Mods.Cnc` each have a generic `Util` class with helpers - it just happens that `OpenRA.Mods.Cnc`'s currently only contains facing-related things.

For the third: We don't do this for anything other fileformat related details, so i'd rather not do it here. URLs in source code don't age well!

Mailaender commented on OpenRA.Mods.Cnc/Util.cs:
```diff
@@ -13,28 +13,42 @@ namespace OpenRA.Mods.Cnc
+		static readonly int[] Facing16 =
+		{
+			8, 24, 40, 56, 72, 88, 104, 120,
+			136, 152, 168, 184, 200, 216, 232, 248
```
This seems to just add +16 at every step. Can you explain why it needs a lookup table?

pchote replied:
Hmm, you appear to be right - a derp on my part. For some reason I had the idea in my head that this was a thing for more than just 32 facings, and I didn't double check this properly. I can remove this and the second commit!

pchote commented:
Fixed and also removed the namespace simplification as a compromise on @penev92's first point and to demonstrate that the second is out of scope of this PR.

pchote committed:
Reimplement ClassicQuantizeFacing using a look-up-table.

penev92 commented:
The namespace identifier is redundant though.

atlimit8 commented:
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/1af90d2c7e051443147611c6e898f15edcba5d41)

atlimit8 commented:
The namespace is needed `Common.Util` from inside `Cnc.Util`.
